Code/Framework/AzFramework/Platform/Mac/AzFramework/Process/ProcessWatcher_Mac.cpp

```diff
@@ -276,21 +278,29 @@ namespace AzFramework
         commandAndArgs[commandTokens.size()] = nullptr;
 
         char** environmentVariables = nullptr;
-        int numEnvironmentVars = 0;
+        
+        AZStd::vector<AZStd::string> allEnvs;
+        for (char **env = ::environ; *env; env++)
+        {
+            allEnvs.push_back(*env);
+        }
+
         if (processLaunchInfo.m_environmentVariables)
         {
-            const int numEnvironmentVars = processLaunchInfo.m_environmentVariables->size();
-            // Adding one more as exec expects the array to have a nullptr as the last element
-            environmentVariables = new char*[numEnvironmentVars + 1];
-            for (int i = 0; i < numEnvironmentVars; i++)
-            {
-                const AZStd::string& envVarString = processLaunchInfo.m_environmentVariables->at(i);
-                environmentVariables[i] = new char[envVarString.size() + 1];
-                environmentVariables[i][0] = '\0';
-                azstrcat(environmentVariables[i], envVarString.size(), envVarString.c_str());
-            }
-            environmentVariables[numEnvironmentVars] = NULL;
+            allEnvs.insert(allEnvs.end(), processLaunchInfo.m_environmentVariables->begin(), processLaunchInfo.m_environmentVariables->end());
+        }
+
+        const int numEnvironmentVars = allEnvs.size();
+        // Adding one more as exec expects the array to have a nullptr as the last element
+        environmentVariables = new char*[numEnvironmentVars + 1];
+        for (int i = 0; i < numEnvironmentVars; i++)
+        {
+            const AZStd::string& envVarString = allEnvs.at(i);
+            environmentVariables[i] = new char[envVarString.size() + 1];
+            environmentVariables[i][0] = '\0';
+            azstrcat(environmentVariables[i], envVarString.size(), envVarString.c_str());
```

Comment:
There is a way to populate the environment variables with allocating any memory by using a fixed_vector

```
constexpr MaxEnvVariables = 128;
using EnvironmentVariableContainer = AZStd::fixed_vector<char*, MaxEnvVariables>;
EnvironmentVariableContainer environmentVariables;
for (char **env = ::environ; *env; env++)
{
    environmentVariables.push_back(*env);
}
for (AZStd::string& processLaunchEnv : *processLaunchInfo.m_environmentVariables)
{
    environmentVariables.push_back(processLaunchEnv.data());
}
environmentVariables.push_back(nullptr);
```

Reply:
Will update.